ckanext/canada/cli.py

```diff
@@ -77,6 +80,61 @@
                 'position', 'metadata_modified']
 
 
+class CaptureExceptionDetails(object):
+    failure_reason = ''
+    failure_trace = ''
+    error = ''
+    has_exceptions = False
+    __reason = None
+    __verbose = False
+    __package_id = None
+    __resource_id = None
+    __view_id = None
```

Comment:
these are class variable which are later shadowed by the instance variables created below, so they're not doing anything here.

Also I strongly recommend against using `__` in front of variables because python's "private" munging behaviour is really weird and tends to just cause problems

But even better would be replacing this whole class with a function using the @contextmanager decorator. That lets you write less code and catch exceptions in the normal python way.

Reply:
@wardi as discussed in our call, I have applied all of the new enclosure context managers with those `nonlocal` variables.

It saved especially a lot of code inside of the add_views and add_datastore methods.